libp2p/__init__.py

```diff
@@ -77,15 +83,22 @@ def new_swarm(
     :param muxer_opt: optional choice of stream muxer
     :param sec_opt: optional choice of security upgrade
     :param peerstore_opt: optional peerstore
+    :param listen_addrs: optional list of multiaddrs to listen on
     :return: return a default swarm instance
     """
     if key_pair is None:
         key_pair = generate_new_rsa_identity()
 
     id_opt = generate_peer_id_from(key_pair)
 
-    # TODO: Parse `listen_addrs` to determine transport
-    transport = TCP()
+    if listen_addrs is None:
+        transport = TCP()
+    else:
+        addr = listen_addrs[0]
+        if addr.__contains__("tcp"):
+            transport = TCP()
+        elif addr.__contains__("quic"):
+            raise ValueError("QUIC not yet supported")
```

Comment:
We'll want an `else` at the bottom here to catch the case where it's not `None`, `tcp`, or `quic`.

Reply:
```suggestion
            raise ValueError("QUIC not yet supported")
        else:
            raise ValueError(f"listen_addrs contains unknown transport: {listen_addrs}")
```